include/aws/cal/private/rsa.h

```diff
@@ -0,0 +1,119 @@
+#ifndef AWS_C_CAL_PRIVATE_RSA_H
+#define AWS_C_CAL_PRIVATE_RSA_H
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include <aws/cal/rsa.h>
+
+#include <aws/common/byte_buf.h>
+#include <aws/common/ref_count.h>
+
+struct aws_rsa_key_pair;
+struct aws_der_decoder;
+
+struct aws_rsa_key_vtable {
+    void (*destroy)(struct aws_rsa_key_pair *key_pair);
+
+    int (*encrypt)(
+        struct aws_rsa_key_pair *key_pair,
+        enum aws_rsa_encryption_algorithm algorithm,
+        struct aws_byte_cursor plaintext,
```

Comment:
something to discuss: while i prefer passing cursors by value, most of the code in this package is passing them by pointer. so should we favor more recent convention or consistency with existing code

Reply:
this is a new private file
I vote we do it the better way

Reply:
coming back to this. 

for public facing apis we already have some inconsistencies in usage of by val vs by ptr, ex. aws_ecc_key_new_from_hex_coordinates is by val and aws_symmetric_cipher_encrypt takes plaintext by val. My vote is to just do it everywhere by val for RSA